app/validators/url_validator.rb

```diff
@@ -0,0 +1,17 @@
+class UrlValidator < ActiveModel::EachValidator
+  def initialize(options)
+    options.reverse_merge!(schemes: %w(http https))
+    options.reverse_merge!(message: _('URL must be valid and scheme must be one of %s') %
+                           options[:schemes].to_sentence)
+    super(options)
+  end
+
+  def validate_each(record, attribute, value)
+    schemes = Array.wrap(options[:schemes]).map(&:to_s)
+    scheme = URI.parse(value).scheme
```

Comment:
This is the crux of the change. It'll check for things like `http://`, `http:`, etc by raising a `URI::InvalidURIError`.